docs/agile-development/collaboration/teaming-up.md

```diff
@@ -0,0 +1,65 @@
+
+This documentation gives a high-level template with some suggestions by aiming to accelerate team swarming phase to achieve a high speed agility however it has no intention to provide a list of "must-do" items.
+
+## Identification
+
+As it's stated in Tuckman's [team phases](https://en.wikipedia.org/wiki/Tuckman%27s_stages_of_group_development), traditional team development has several stages.
+However those phases can be extremely fast or sometimes mismatched in CSE engagement teams due to external factors.
+
+In order to minimize the risk and set the exceptations on the right way for all parties, an identification phase is important to understand each other.
+Some potential steps in this phase may be as following (not limited):
+
+- [Working agreement](../team-agreements/working-agreements/readme.md)
+
+- Identification of styles/preferences in communication, sharing, learning, decision making of each team member
+  
+  - Talking about necessity of pair programming
+  - Decisions on backlog management & grooming meetings, weekly design sessions, social time sessions...etc.
+  - Sync/Async communication methods, work hours/flexible times
+
+- Decisions and identifications of charts that will be helpful to provide transparent and true information to everyone
+
+- Identification of "Software Craftspersonship" areas which means the tools and methods will be widely used during the engagement and taking the required actions on team upskilling side if necessary.
+
+  - Github, VSCode LiveShare, AzDevOps, necessary development tools & libraries ... more.
+
+- Identification of communication channels, feedback loops and recurrent team call slots out of regular sprint meetings
+
+- Introduction to [Technical Agility Team Manifesto](../team-agreements/team-manifesto/readme.md) and planning the technical delivery by aiming to keep
+technical debt risk minimum.
+
+## Following the Plan and Agile Debugging
+
+Identification phase accelerates the process of building a safe environment for every individual in the team, later on team has the required assets to follow the plan.
+And it is team's itself responsibility (SEs, TPM, Lead) to debug their Agility level.
+
+In every team stabilization takes time and pro-active agile debugging is the best accelerator to decrease the distraction away from sprint/engagement goal.
+
+Just as an example, agility debugging activities may include:
+
+- Dashboards related with "Goal" such as burndown/burnout, Item/PR Aging, Mood Chart ..etc. are accessible to the team and team is always up-to-date
+
+- Backlog Grooming meetings
+  - Size of stories (Too big? Too small?)
+  - Are US and tasks clear ?
+  - Are Acceptance Criterias enough and right?
+  - Is everyone ready-to-go after taking the User Story/Task?
+
+- Running [Efficient Retrospectives](../retrospectives/readme.md)
```

Comment:
Please consider moving the questions to the retrospectives document so the Retrospective section becomes the source of truth for the practice.

Reply:
thank you very much for this suggestion :) 
I will do follow-ups on it.